src/constants/textConstants.js

```diff
@@ -1,13 +1,55 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+
 import keyMirror from 'keymirror';
+import { helpers } from '../utils/';
 
 const TEXT_TAGS = keyMirror({
+  condensed: null,
   span: null,
   p: null,
   strong: null
 });
 
+const CondensedText = ({
+  children,
+  className,
+  enableCopyIcon = false,
+  prefix = 4,
+  style,
+  suffix = 4,
+  theme: { text: { iconActive, iconInactive }, glyphs: { copyIcon } }
+}) => (
+  <React.Fragment>
+    <span
+      className={`
+        ${enableCopyIcon ? iconActive : iconInactive}
+        ${className} bpanel-ui-text-condensed
+      `}
+      onClick={() => !enableCopyIcon && helpers.onCopy(children)}
+      style={style}
+      title={`${children}`}
+    >
+      {helpers.condenseText(children, prefix, suffix)}
+    </span>{' '}
+    {enableCopyIcon && (
+      <i
+        className={`fa fa-copy ${copyIcon}`}
+        onClick={() => helpers.onCopy(children)}
+      />
+    )}
+  </React.Fragment>
+);
+
+CondensedText.propTypes = {
+  children: PropTypes.string.isRequired,
+  enableCopyIcon: PropTypes.bool,
+  prefix: PropTypes.number,
+  suffix: PropTypes.number
+};
+
 const TEXT_ELEMENTS = {
+  [TEXT_TAGS.condensed]: CondensedText,
```

Comment:
Does this not need to be able to have props passed?

Reply:
Props are passed into ```TextElement```.  In this case ```CondensedText``` is the function which destructures the props and generates the fragment. ```Props``` are being passed, just implicitly. 

The following invocation patterns work exactly the same as current implementation  : 
```
const TEXT_ELEMENTS = {
  [TEXT_TAGS.condensed]: CondensedText,
  [TEXT_TAGS.span]: props => React.createElement('span'),
  [TEXT_TAGS.p]: props => React.createElement('p'),
  [TEXT_TAGS.strong]: props => React.createElement('strong')
};
```



How about moving ```CondensedText``` to ```Text/TextEffects.js``` with the same pattern as our other standard elements (header and text constants) and roll them all into one ```Text.js``` object.

Here is what your directory will look like:

![image](https://user-images.githubusercontent.com/18730408/43740905-8177a21e-9981-11e8-997b-b077239ec314.png)


Thoughts?

@tynes @bucko13

Reply:
I like the idea of `Text/TextEffects`

Trying to think of other `TextEffects`, it does leave it open to many different types in the future

Reply:
I like this idea too. My only question would be if other text effects would be rolled into the same component or separated out? Either way, since this is all self-contained we'd be able to update without breaking anything that depends on it.